libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -226,11 +226,11 @@ void WaveClip::OnProjectTempoChange(
       // read-up or signal analysis) we can use something smarter than that. In
       // the meantime, use the tempo of the project when the clip is created as
       // source tempo.
-      mRawAudioTempo = oldTempo;
+      mRawAudioTempo = oldTempo.value_or(newTempo);
```

Comment:
I don't remember what this line fixes -- was it only necessary when you were using temporary side effects on clips?

Reply:
Not sure anymore either, but it still is better this way: `WaveClip::OnProjectTempoChange` is the only way a clip's project tempo can be set, hence `WaveTrack::InsertClip` calling it with `oldTempo == nullopt`.
Best would be to add a dedicated API, but I'm lazy.